keras_cv/layers/preprocessing/equalization.py

```diff
@@ -52,81 +53,104 @@ def __init__(self, value_range, bins=256, **kwargs):
+    def equalize_channel(self, images, channel_index):
         """equalize_channel performs histogram equalization on a single channel.
-
+    
         Args:
             image: int Tensor with pixels in range [0, 255], RGB format,
                 with channels last
             channel_index: channel to equalize
         """
-        image = image[..., channel_index]
+        images = images[..., channel_index]
         # Compute the histogram of the image channel.
-        histogram = tf.histogram_fixed_width(image, [0, 255], nbins=self.bins)
-
+        partial_hist = partial(tf.histogram_fixed_width, value_range=[0, 255], nbins=self.bins)
+        histogram = tf.vectorized_map( # TODO: bottleneck
+            partial_hist, images, fallback_to_while_loop=True, warn=True
+        )
+    
         # For the purposes of computing the step, filter out the non-zeros.
         # Zeroes are replaced by a big number while calculating min to keep
         # shape constant across input sizes for compatibility with
         # vectorized_map
-
+    
         big_number = 1410065408
         histogram_without_zeroes = tf.where(
             tf.equal(histogram, 0),
             big_number,
             histogram,
         )
-
+    
         step = (
-            tf.reduce_sum(histogram) - tf.reduce_min(histogram_without_zeroes)
+            tf.reduce_sum(histogram, axis=-1) - tf.reduce_min(histogram_without_zeroes, axis=-1)
         ) // (self.bins - 1)
-
+    
         def build_mapping(histogram, step):
             # Compute the cumulative sum, shifting by step // 2
             # and then normalization by step.
-            lookup_table = (tf.cumsum(histogram) + (step // 2)) // step
+            _step = tf.where(
+                tf.equal(step, 0),
+                1, # replace where step is 0 with 1 to avoid division by 0. This doesn't change anything because in the tf.where we return the original image where step == 0
+                step,
+            )
+            _step = tf.expand_dims(_step, -1)
+            bacth_size = tf.shape(histogram)[0]
+            lookup_table = (tf.cumsum(histogram, axis=-1) + (_step // 2)) // _step
             # Shift lookup_table, prepending with 0.
-            lookup_table = tf.concat([[0], lookup_table[:-1]], 0)
+            lookup_table = tf.concat([tf.tile([[0]], [bacth_size, 1]), lookup_table[..., :-1]], 1)
             # Clip the counts to be in range. This is done
             # in the C code for image.point.
             return tf.clip_by_value(lookup_table, 0, 255)
 
         # If step is zero, return the original image. Otherwise, build
         # lookup table from the full histogram and step and then index from it.
-        result = tf.cond(
-            tf.equal(step, 0),
-            lambda: image,
-            lambda: tf.gather(build_mapping(histogram, step), image),
+        # Actually, the lookup table is built for all the images, no matter which is the corresponding value of step
```

Comment:
Let us update the comment here 
"Actually, the lookup table is built for all the images, no matter which is the corresponding value of step
This is an extra computation, but allows to do everything vectorized and avoiding division by 0"
->
The lookup table is built for all images, regardless of the corresponding value of step.